Sources/_StringProcessing/Engine/MEBuilder.swift

```diff
@@ -142,8 +142,8 @@ extension MEProgram.Builder {
     instructions.append(.init(.clearThrough))
     fixup(to: t)
   }
-  mutating func buildFail() {
-    instructions.append(.init(.fail))
+  mutating func buildFail(preservingCaptures: Bool = false) {
+    instructions.append(.init(.fail, .init(bool: preservingCaptures)))
```

Comment:
What would you think about renaming `fail` to `backtrack` or something similar? I always have to remember that the fail instruction isn't an overall failure, just a failure of the immediate execution track.

Reply:
Sure, if we can do it pervasively and consistently